cosmic-settings/src/pages/bluetooth/backend/dbus.rs

```diff
@@ -0,0 +1,322 @@
+use bluez_zbus::get_adapter;
+use futures::join;
+use zbus::Connection;
+
+use super::{Active, Adapter, Device};
+
+pub async fn get_devices<'a>() -> Result<Vec<Device>, zbus::Error> {
+    let connection = match Connection::system().await {
+        Ok(c) => c,
+        Err(e) => {
+            tracing::error!("zbus connection failed. {e}");
+            return Err(e);
+        }
+    };
```

Comment:
It would be good if each function accepts an already established connection. They're clonable so they can be freely passed around.

Reply:
I have refactored the code and done so when possible. I've made the choice not to do so when moving in `tokio::spawn` for two reasons:
- Cleaner code - no need to make variable duplication + struct meber
- Re-instantiating the connection when needed will handle re-connection (e.g DBus crash)